qutebrowser/completion/models/urlmodel.py

```diff
@@ -86,6 +88,10 @@ def url(*, info):
         models['bookmarks'] = listcategory.ListCategory(
             'Bookmarks', bookmarks, delete_func=_delete_bookmark, sort=False)
 
+    if buffers and 'buffers' in categories:
+        models['buffers'] = listcategory.ListCategory(
```

Comment:
I haven't looked into it too much, but I think that some of the work in [miscmodels](http://www.github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/completion/models/miscmodels.py#L125) can be re-used here (avoiding the functions added to utils). You would also get the delete function for free as well.

However, you would need to find a solution to avoid breaking each window into its own category

Reply:
I've been trying to figure out how to reuse the `miscmodels` work, but the biggest problem I see right now is that the tuples are in different order. I've been toying with the idea of having a lambda transformer to get the tuples into the right shape for this function.

Thoughts?